metals/src/main/scala/scala/meta/internal/metals/codeactions/BracelessBracefulSwitchCodeAction.scala

```diff
@@ -0,0 +1,558 @@
+            applyWithSingleFunction
+          )
+      else None
+
+    val result =
+      for {
+        tree <- maybeTree
+        document <- buffers.get(path)
+      } yield {
+        tree match {
+          case _: Pkg => None.toSeq
+          case classDefn: Defn.Class =>
+            createCodeActionForTemplateHolder(
+              classDefn,
+              path,
+              document,
+              classDefn.templ,
+              "class definition"
+            ).toSeq
+          case objectDefn: Defn.Object =>
+            createCodeActionForTemplateHolder(
+              objectDefn,
+              path,
+              document,
+              objectDefn.templ,
+              "object definition"
+            ).toSeq
+          case traitDefn: Defn.Trait =>
+            createCodeActionForTemplateHolder(
+              traitDefn,
+              path,
+              document,
+              traitDefn.templ,
+              "trait definition"
+            ).toSeq
+          case enumDefn: Defn.Enum =>
+            createCodeActionForTemplateHolder(
+              enumDefn,
+              path,
+              document,
+              enumDefn.templ,
+              "enum definition"
+            ).toSeq
+          case valDefn: Defn.Val =>
+            createCodeActionForPotentialBlockHolder(
+              valDefn,
+              path,
+              document,
+              valDefn.rhs,
+              "val definition"
+            ).toSeq
+          case varDefn: Defn.Var =>
+            varDefn.rhs.flatMap {
+              createCodeActionForPotentialBlockHolder(
+                varDefn,
+                path,
+                document,
+                _,
+                "var definition"
+              )
+            }.toSeq
+          case defDefn: Defn.Def =>
+            createCodeActionForPotentialBlockHolder(
+              defDefn,
+              path,
+              document,
+              defDefn.body,
+              "def definition"
+            ).toSeq
+          case termTry: Term.Try =>
+            Seq(
+              createCodeActionForPotentialBlockHolder(
+                termTry.expr,
+                path,
+                document,
+                termTry.expr,
+                "try expression"
+              ),
+              createCodeActionForCatchP(
+                termTry,
+                path,
+                document,
+                "catch expression"
+              ),
+              termTry.finallyp.flatMap(finallyp =>
+                createCodeActionForPotentialBlockHolder(
+                  finallyp,
+                  path,
+                  document,
+                  finallyp,
+                  "finally expression"
+                )
+              )
+            ).flatten
+          case caseTree: Case =>
+            createCodeActionForCasesListWrapper(caseTree, path).toSeq
+          case termIf: Term.If =>
+            Seq(
+              createCodeActionForPotentialBlockHolder(
```

Comment:
Same as with Try, could we only show the most relevant action for the current position?

Reply:
there is not a straightforward way to know whether we are at then/else branches. It only captures the whole if-else tree. Any suggestions for capturing where we are at the tree are welcome

Reply:
I would just check if we are closer to thenp or elsep statement.

Reply:
Could be! I would check! Thank you!